cpp/src/parquet/arrow/reader.h

```diff
@@ -116,15 +116,14 @@ class RowGroupReader;
 class PARQUET_EXPORT FileReader {
  public:
   /// Factory function to create a FileReader from a ParquetFileReader and properties
-  static ::arrow::Status Make(::arrow::MemoryPool* pool,
-                              std::unique_ptr<ParquetFileReader> reader,
-                              const ArrowReaderProperties& properties,
-                              std::unique_ptr<FileReader>* out);
```

Comment:
Could you keep the `arrow::Status` version with deprecated mark so that users will be able to migrate to the `arrow::Result` version smoothly?
See also: https://github.com/apache/arrow/pull/44785

Reply:
Got it. Will add comments instead and address the comments. Thanks @kou

Reply:
+1. We need to mark the existing ones as deprecated without deleting them.